### ISSUE TITLE
OffscreenGibFader: Null ref when transitioning rooms?

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#320]
Found in 0.0.15d. Player was knocked back into a corridor from a grenade.

Error log mentions `Gib.RemoveFromNetIDDatabaseAndReturnToPool`:

```csharp
public void RemoveFromNetIDDatabaseAndReturnToPool () {
	gibStruct.parentNetIDDatabase.RemoveGibStruct(gibStruct.netID);
	game.dungeonRoom.instantiatedGibs.Remove(this);
	ReturnGibToContainer();
}
```
`gibStruct` is probably null from the OffscreenGibFader class.

![image](https://user-images.githubusercontent.com/280814/66753170-b010db80-eeef-11e9-9285-585edc38f2fe.png)

**Comment by ChrisNZL** [comment]
Similar report from 0.0.18 (iOS version):

```
4:30:11, Frame 6746, EXCEPTION »  NullReferenceException: Object reference not set to an instance of an object.
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

T2.Gib.RemoveFromNetIDDatabaseAndReturnToPool ()
T2.DebrisFaderComponent.Update ()

GameStates: RompingThroughDungeon
GameSetupMode: Singleplayer
Players: 1
Room: 20 / 20
RoomModifiers: None
```

**Comment by ChrisNZL** [comment]
Should be fixed in 0.1.0.

Added a null check against `gibStruct` within `Gib.RemoveFromNetIDDatabaseAndReturnToPool`.
